docs/REST API/rest-api.md

```diff
@@ -8979,49 +8986,156 @@ Request (using `filters`):
 }
 ```
 
-### Get batch script summary
+### List batch scripts
 
-Get statuses and host counts for a batch-run script.
+Returns a list of batch script executions.
 
-`GET /api/v1/fleet/scripts/batch/summary/:batch_execution_id`
+`GET /api/v1/fleet/scripts/batch`
 
 #### Parameters
 
 | Name            | Type    | In   | Description                                                                                    |
 | ----            | ------- | ---- | --------------------------------------------                                                   |
-| batch_execution_id | string | path | **Required**. The ID returned from a batch script run. |
+| team_id         | integer | query | _Available in Fleet Premium_. Filters to batch script runs for the specified team. |
+| status          | string  | query | Filters to batch script runs with this status. Either `"started"`, `"scheduled"`, or "`completed`". |
+| page            | integer | query | Page number of the results to fetch. |
+| per_page        | integer | query | Results per page. |
 
 
 #### Example
 
-`GET /api/v1/fleet/scripts/batch/summary/abc-def`
+`GET /api/v1/fleet/scripts/batch`
 
 ##### Request body
 
 ```json
 {
-  "batch_execution_id": "e797d6c6-3aae-11ee-be56-0242ac120002"
+  "team_id": 123,
+  "status": "completed"
+}
+```
+
+##### Default response
+
+`Status: 200`
+
+```json
+{
+  "batch_executions": [
+    {
+      "script_id": 555,
+      "script_name": "my-script.sh",
+      "team_id": 123,
+      "not_before": "2025-07-01T15:00:00Z",
+      "completed_at": "2025-07-06T15:00:00Z",
+      "status": "completed",
```

Comment:
If a batch has `"status": "started"`, can `"not_before"` be `null` in the case that the batch was run immediately? Seems like this should be the case, so seems like we also need a `"started_at"` field right?

Reply:
If we do keep `status`, seems `canceled` should be an option and not its own field, though better to remove `status` and update to `canceled_at`.

Reply:
sgtm

Reply:
The `canceled_at` / `completed_at` approach dovetails with the "status" of a summary being completely defined by (not_before, started_at,  completed_at, canceled_at) and not having a "status" field.

This suggestion dovetails with the current, "status"-field based approach. Both make sense to me, so given we want to keep `status` this proposal makes a lot of sense

Reply:
decision:
`completed_at` –> `finished_at`
`status.completed` -> `status.finished`
`canceled` flag remains